pkg/dns/dns.go

```diff
@@ -643,31 +646,37 @@ func (kd *KubeDNS) getClusterZone() (string, error) {
 		// TODO(madhusudancs): Move this to external/v1 API.
 		nodeList, err := kd.kubeClient.Core().Nodes().List(kapi.ListOptions{})
 		if err != nil || len(nodeList.Items) == 0 {
-			return "", fmt.Errorf("failed to retrieve the cluster nodes: %v", err)
+			return "", "", fmt.Errorf("failed to retrieve the cluster nodes: %v", err)
 		}
 
 		// Select a node (arbitrarily the first node) that has `LabelZoneFailureDomain` set.
 		for _, nodeItem := range nodeList.Items {
-			if _, ok := nodeItem.Labels[unversioned.LabelZoneFailureDomain]; !ok {
+			_, zfound := nodeItem.Labels[unversioned.LabelZoneFailureDomain]
+			_, rfound := nodeItem.Labels[unversioned.LabelZoneRegion]
+			if !zfound || !rfound {
 				continue
 			}
 			// Make a copy of the node, don't rely on the loop variable.
 			node = &(*(&nodeItem))
```

Comment:
for a future PR? - When we find the first appropriate node, should we not break out of the for loop? (comment in line 652 indicates that we are just picking the first node, code indicates we are picking the last appropriate node)